public/components/userpage.js

```diff
@@ -0,0 +1,100 @@
+
+ajaxRequest("GET", window.location + "?ajax=true", {}, {}, renderComponent);
```

Comment:
Because with the header there's the problem with Chrome, with a query it works perfectly.
I'd change it to a `query` everywhere else to be honest, it's done in no time.